File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
answer_sheet = {'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'}
def solution(s):
    answer = ''
    tmp = ''
    for i in s:
        if i.isdigit():
            answer += i
        else:
            tmp += i
            if answer_sheet.get(tmp, 0) != 0:
                answer += answer_sheet[tmp]
                tmp = ''
        
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <cctype>

using namespace std;

int answer_sheet(string s) {
    if (s == "zero") return 0;
    else if (s == "one") return 1;
    else if (s == "two") return 2;
    else if (s == "three") return 3;
    else if (s == "four") return 4;
    else if (s == "five") return 5;
    else if (s == "six") return 6;
    else if (s == "seven") return 7;
    else if (s == "eight") return 8;
    else if (s == "nine") return 9;
    else return -1;
}

int solution(string s) {
    string answer = "";
    
    string tmp="";
    for (char c: s) {
        if (isdigit(c)) {
            answer += c;
            continue;
        }
        tmp += c;
        if (tmp.size() > 2) {
            int ck = answer_sheet(tmp);
            if (ck > -1) {
                tmp = "";
                answer += '0' + ck;
            }
        }
    }
    
    int res = stoi(answer);
    
    return res;
}